packages/next/src/server-build.ts

```diff
@@ -2089,6 +2089,19 @@ export async function serverBuild({
       // with that routing section
       ...afterFilesRewrites,
 
+      // Ensure that after we normalize `afterFilesRewrites`, unmatched actions are routed to the correct handler
+      // e.g. /foo/.action -> /foo.action. This should only ever match in cases where we're routing to an action handler
+      // and the rewrite normalization led to something like /foo/$1$rscsuff, and $1 had no match
+      ...(hasActionOutputSupport
+        ? [
+            {
+              src: `${path.posix.join('/', entryDirectory, '(.+)/\\.action$')}`,
+              dest: `${path.posix.join('/', entryDirectory, '$1.action')}`,
+              check: true,
```

Comment:
Since Next.js doesn't emit any `.action` outputs and thus won't exist on the filesystem, would this make more sense? cc @ijjk 

```suggestion
              continue: true,
```